csrc/scheduler/matmul_utils.cpp

```diff
@@ -232,12 +229,6 @@ std::string isMatmulFusionDefinitionSupported(
 
   // Fusion topology check
   {
-    const auto& roles_map_opt = mma_utils::getTensorsRoles(fusion, props);
-    if (!roles_map_opt.isValid()) {
-      return roles_map_opt.getErrorMsg();
-    }
-
-    const auto& roles_map = roles_map_opt.getData();
     auto entry = roles_map.find(MatmulRole::INPUT_A);
     std::set<TensorView*> tvs_with_roles;
```

Comment:
I know this is not added by this PR, but it doesn't look like we need the actual tensorviews, just the total count.

Reply:
The count of the unique tensorview is needed I think. I think this code might be for handling cases where a tensor has multiple roles, for example `A@A`. In that case count is not sufficient to determine whether there are TVs without roles.